lms/templates/courseware/course_about.html

```diff
@@ -279,177 +281,182 @@ <h2 class="custom-h2">
-                      <div class="important-dates-item-text final-date">${course_end_date}</div>
-                  % else:
-                    <%
-                        course_date_string = course_end_date.strftime('%Y-%m-%dT%H:%M:%S%z')
-                    %>
-                    <span class="important-dates-item-text final-date localized_datetime" data-language="${LANGUAGE_CODE}" data-format-language="${LANGUAGE_CODE}" data-format="ISODate" data-datetime="${course_date_string}"></span>
+
+                  <hr class="course__details__line">
+
+                  % if (course.enrollment_start or course.enrollment_end):
+                  <%course_start_enrollment = course.enrollment_start%>
+                  <%course_end_enrollment = course.enrollment_end%>
+                  <%dates_enrollment = [course_start_enrollment, course_end_enrollment]%>
+                  <%now = datetime.datetime.now()%>
+                  <%test = datetime.datetime.now(pytz.utc)%>
+                  <%dni = (relativedelta(course_end_enrollment,test))%>
+                  <li class="list-group-item">
+                    <div class="course-details-icon course__details__margin">
+                      <i class="fas fa-bell" aria-hidden="true"></i>
+                    </div>
+                    <div class="course-details-content">
+                      <h6>${_("Enrollment dates")}:</h6>
+                      % if isinstance(dates_enrollment, string_types):
+                      <div class="important-dates-item-text final-date f-14 h-19">${course_start_enrollment} - ${course_end_enrollment}</div>
+                      % else:
+                        <span class="sr-only">${_("from")}</span>
+                      <%course_start_string_enrollment = course_start_enrollment.strftime('%Y-%m-%dT%H:%M:%S%z')%>
+                        <span class="important-dates-item-text final-date localized_datetime f-14 h-19" data-language="${LANGUAGE_CODE}" data-format-language="${LANGUAGE_CODE}" data-format="PlDate" data-datetime="${course_start_string_enrollment}"></span>
+                        <span class="f-14 h-19" aria-hidden="true">&nbsp;-&nbsp;</span>
+                        <span class="sr-only">${_("to")}</span>
+                      <%course_end_string_enrollment = course_end_enrollment.strftime('%Y-%m-%dT%H:%M:%S%z')%>
+                        <span class="important-dates-item-text final-date localized_datetime f-14 h-19" data-language="${LANGUAGE_CODE}" data-format-language="${LANGUAGE_CODE}" data-format="PlDate" data-datetime="${course_end_string_enrollment}"></span>
+                      % endif
+                      <div class="d-flex">
+                      <%difference_in_years = dni.years%>
+                      <%difference_in_months = dni.months%>
+                      <%difference_in_days = dni.days%>
+                      <%difference_in_hours = dni.hours%>
+                      % if (difference_in_years  != 0):
```

Comment:
Czy nie prościej by było coś w tym stylu? 
"Enrollment ends in: " ungettext("{mont} month", "{month} months", int(difference_in_months)).format(month=difference_in_months)) if difference_in_months else '' + ungettext("{day} day", "{day} days", int(difference_in_days)).format(day=difference_in_days)) if difference_in_days else '' +  .... ,
oczywiście w składni mako

Reply:
miałam przygotować aby pozostały czas zapisu na kurs był wyświetlany tak:
 "> rok: rok + miesiące (1 rok 3 mies.)
< rok: miesiące + dni (5 mies. 5 dni) 
< miesiąca: dni + godziny (20 dni 3 godz.) 
Poniżej tygodnia prosiłabym dodatkowo, aby kolor całej frazy "koniec zapisów za: xxx" zmieniał się na kolor czerwony.
< tygodnia: dni + godziny (5 dni 8 godz.)
< 2 dni: dzień + godziny (1 dzień 15 godz.)
< dnia: godziny + minuty (12 godz. 45 min.) 
<godziny: minuty (15 min.) 
Prosiłabym także, aby liczba zero została pominięta, tzn. jeśli będzie "rok 0 mies." lub "3 dni 0 godz." to żeby "0" zostało pomięte i aby została tylko informacja "rok" / "3 dni". "

przerobiłam ten fragment, teraz jest chyba czytelniejszy??